lib/dor/text_extraction/abbyy/results.rb

```diff
@@ -12,11 +12,21 @@ class Results
         # ab123cd4567.xml.0001.result.xml
         RESULT_FILE_PATTERN = /(?<druid>\w+)\.xml(?:\.(?<run_index>\d+))?\.result\.xml/
 
-        def initialize(result_path:)
+        # Find the latest Abbyy results XML file by druid, since there coule be more than one.
+        # @param druid [String] The druid to use to look for results
+        # @return [Results, nil] The Results object or nil, if one wasn't found.
+        def self.find_latest(druid:, logger: nil)
+          bare_druid = DruidTools::Druid.new(druid).id
+          result_files = Dir.glob("#{Settings.sdr.abbyy.local_result_path}/#{bare_druid}*.xml")
```

Comment:
does `Dir.glob` automatically return the results with the most recent first? if not, you could sort them by `run_index` to get the most recent (highest index)

Reply:
Yeah, I only realized that it did because my editor told me (maybe rubcop?). 

> If optional keyword sort is given, its value specifies whether the array is to be sorted; the default is true. Passing value false with that keyword disables sorting (though the underlying file system may already have sorted the array).

I think it's worth a comment explaining it, thanks for the nudge.

Reply:
Oh wait, actually you're right their sorted in the ascending order. So this is actually a bug.

Reply:
Ok I've updated it to get the last one in the list. Hopefully this won't really become needed in practice since we will be removing the result.xml once the workflow completes.